### ISSUE TITLE
gradient checkpointing and caching

### ISSUE THREAD

**Issue by haorannlp** [bigcode-project/starcoder#112]
In this line of code 
https://github.com/bigcode-project/starcoder/blob/d72c7fe3dda81d47ad9b851f9567393fb6b551b9/finetune/finetune.py#L82
The value when specifying `no_gradient_checkpoint` is `False`, 
while when initializing the model at https://github.com/bigcode-project/starcoder/blob/d72c7fe3dda81d47ad9b851f9567393fb6b551b9/finetune/finetune.py#L243 
it is always using caching. Should `use_cache=False` when using gradient_checkpointing?

**Comment by lvwerra** [comment]
This seems to be indeed a bug but since `args.no_gradient_checkpointing` is actually never used and we does don't to gradient checkpointing it makes no difference. It only matters when gradient checkpointing is activated.
